### PR TITLE
Add test to verify that NamespaceValidator will reject request if namespace name from task does not match name from request

### PR DESCRIPTION
**Pull request by ast2023**
**What changed?**
one test case added

**Why?**
cover requirement

**How did you test it?**
run test, made sure that when assertion is incorrect the test will fail

**Potential risks**
no impact

**Is hotfix candidate?**
No

### CODE REVIEW

**Review by tdeebswihart — APPROVED** [review]

**Review by dnr — COMMENTED** [review]
please keep the headers in the PR description so it's understandable without memorizing the template

File: tests/namespace_interceptor_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+	err := sut.StartWorkflowExecution(s.namespace)
+	s.NoError(err)
+
+	// start workflow task
+	err = sut.PollWorkflowTaskQueue(s.namespace)
+	s.NoError(err)
+
+	err = sut.RespondWorkflowTaskCompleted("another-namespace")
+	s.Errorf(err, "Invalid request was processed, make sure request validator interceptor is loaded")
```
Comment (alexshtin) [inline tests/namespace_interceptor_test.go:52]:
```suggestion
	s.Error(err, "Invalid request was processed successfully, make sure NamespaceRequestValidator interceptor is used")
```

File: tests/namespace_interceptor_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	taskQueue       *taskqueuepb.TaskQueue
+	stickyTaskQueue *taskqueuepb.TaskQueue
+	id              string
+	taskToken       []byte
+}
+
+func newSystemUnderTestConnector(s *functionalSuite) *sutConnector {
+	id := uuid.New()
+	taskQueue := &taskqueuepb.TaskQueue{Name: id, Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	stickyTaskQueue := &taskqueuepb.TaskQueue{Name: "test-sticky-taskqueue", Kind: enumspb.TASK_QUEUE_KIND_STICKY, NormalName: id}
```
Comment (alexshtin) [inline tests/namespace_interceptor_test.go:70]: I would move all those inline.

File: tests/namespace_interceptor_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		suite:           s,
+		identity:        "worker-1",
+		taskQueue:       taskQueue,
+		stickyTaskQueue: stickyTaskQueue,
+		id:              id,
+	}
+
+}
+
+func (b *sutConnector) StartWorkflowExecution(namespace string) error {
```
Comment (alexshtin) [inline tests/namespace_interceptor_test.go:82]: All these methods don't need to be exported. I.e.:
```suggestion
func (b *sutConnector) startWorkflowExecution(namespace string) error {
```

Comment (alexshtin) [reply tests/namespace_interceptor_test.go:82]: Also I am trying to use special `namespace.Name` and `namespace.ID` types everywhere to pass namespace name and Id.

**Review by alexshtin — APPROVED** [review]

File: tests/namespace_interceptor_test.go (hunk 3 above)
Comment (ast2023) [reply tests/namespace_interceptor_test.go:82]: 👍

File: tests/namespace_interceptor_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	s.NoError(err)
+
+	err = sut.RespondWorkflowTaskCompleted("another-namespace")
+	s.Errorf(err, "Invalid request was processed, make sure request validator interceptor is loaded")
+
+	err = sut.RespondWorkflowTaskCompleted(s.namespace)
+	s.NoError(err, "Valid request was rejected")
+}
+
+type sutConnector struct {
```
Comment (dnr) [inline tests/namespace_interceptor_test.go:58]: all these top-level names should be renamed to something more specific to this test.. these names are package-wide so they're likely to conflict with other tests.

I also don't see a problem with inlining all of these requests, rpc calls, etc in a single function here. that's the predominant style with these tests.

Comment (ast2023) [reply tests/namespace_interceptor_test.go:58]: @dnr thank you for the suggestion. How much time did it take to read and understand what test is doing?

**Pull request merged**